src/com/rath/rathbot/cmd/disc/actions/UnbanCmd.java

```diff
@@ -1,6 +1,100 @@
+
 package com.rath.rathbot.cmd.disc.actions;
 
+import com.rath.rathbot.RathBot;
+import com.rath.rathbot.cmd.PermissionsTable;
+import com.rath.rathbot.cmd.RBCommand;
+
+import sx.blah.discord.api.IDiscordClient;
+import sx.blah.discord.handle.obj.IMessage;
+import sx.blah.discord.handle.obj.IUser;
 
-public class UnbanCmd {
+/**
+ * Unbans a user by their UID or @mention.
+ * 
+ * @author nlehenba
+ * @author Tim Backus tbackus127@gmail.com
+ *
+ */
+public class UnbanCmd extends RBCommand {
+  
+  @Override
+  public String getCommandName() {
+    return "unban";
+  }
+  
+  @Override
+  public String getCommandDescription() {
+    return "(Mod only) Unbans a user defined by their UID or @mention.";
+  }
+  
+  @Override
+  public String getCommandUsage() {
+    return "rb! unban <uid|mention>";
+  }
+  
+  @Override
+  public int permissionLevelRequired() {
+    return RBCommand.PERM_MODERATOR;
+  }
+  
+  @Override
+  public boolean requiresDirectMessage() {
+    return false;
+  }
   
+  @Override
+  public boolean executeCommand(final IMessage msg, final String[] tokens, final int tokDepth) {
+    
+    // Ensures at least minimum valid arguments used.
+    if (tokens.length < 3) {
+      RathBot.sendMessage(msg.getChannel(), "Syntax Error! Usage: rb! kick <uid>");
+      return RBCommand.STOP_CMD_SEARCH;
+    }
+    
+    // Tests if the first argument of the command is an @mention or uid, then processes the argument accordingly.
+    long unbanUserID = 0;
+    String userToken = tokens[tokDepth + 1];
+    if (userToken.matches("<@!?\\d+>")) {
+      
+      int hasNickName = 0;
+      
+      if (userToken.matches("<@!\\d+>")) hasNickName = 1;
+      
+      // If argument is @mention, substring to get UID
+      try {
+        unbanUserID = Long.parseLong(
+            userToken.substring((userToken.indexOf('@') + hasNickName + 1), userToken.indexOf('>')));
+      } catch (NumberFormatException nfe) {
+        nfe.printStackTrace();
+      }
+    } else {
+      
+      // Else argument is UID, parse UID
+      try {
+        unbanUserID = Long.parseLong(tokens[tokDepth + 1]);
+      } catch (NumberFormatException nfe) {
+        nfe.printStackTrace();
+      }
+    }
+    
+    final IDiscordClient client = RathBot.getClient();
+    
+    // Create IUser object from unbanUserID to reference user in messages.
+    IUser unbannedUser = client.getUserByID(unbanUserID);
+    if (unbannedUser == null) {
```

Comment:
Perhaps we should also make the MessageHelper.getUIDByToken() method also check the UID against the client.

Reply:
Again, agreed. Parsing the UID from a UID *OR* @mention token is used in a lot of commands and even commands that can't use @mention will still benefit from this whole if else block.